webrender/res/ps_border_corner.glsl

```diff
@@ -336,28 +339,39 @@ void main(void) {
     if (all(lessThan(local_pos * vClipSign, vClipCenter * vClipSign))) {
         vec2 p = local_pos - vClipCenter;
 
+        // The coordinate system is snapped to pixel boundaries. To sample the distance,
+        // however, we are interested in the center of the pixels which introduces an
+        // error of half a pixel towards the exterior of the curve (See issue #1750).
+        // This error is corrected by offsetting the distance by half a device pixel.
+        // This not entirely correct: it leaves an error that varries between
+        // 0 and (sqrt(2) - 1)/2 = 0.2 pixels but it is hardly noticeable and is better
+        // than the constant sqrt(2)/2 px error without the correction.
+        // To correct this exactly we would need to offset p by half a pixel in the
+        // direction of the center of the ellipse (a different offset for each corner).
+
+        // A half device pixel in css pixels (using the average of width and height in case
+        // there is any kind of transform applied).
+        float half_px = 0.25 * (fw.x + fw.y);
```

Comment:
it sounds to me that `0.5 * length(fw)` would be slightly more appropriate, which is almost similar to `aa_step`

Reply:
if you consider fw to be "kinda like a device pixel ratio in x and y", `length(fw)` gives something that is the diagonal of the rectangle described as the pixel ratio whereas here we want something that is equal to a side of that square (the aa math here makes the approximation that fw.x and fw.y are somewhat similar). There is an average of the two as an effort to not do something bad with some transforms.
Since this is all on top of approximations I first felt like it wouldn't matter too much and tried to just use `0.5*length(fw)` (even if it overshoots a little), and the results weren't as good so I ended up with this instead.